app/res/layout/create_pin_view.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:id="@+id/pin_prompt_text"
+        android:layout_centerHorizontal="true"
+        android:layout_margin="@dimen/standard_spacer_large"
+        android:textSize="@dimen/text_large" />
+
+    <EditText
+        android:id="@+id/pin_entry"
+        android:layout_height="60dp"
+        android:layout_width="match_parent"
+        android:textSize="@dimen/text_largest"
+        android:layout_margin="@dimen/standard_spacer_large"
+        android:layout_centerVertical="true"
+        android:gravity="center_horizontal"
+        android:inputType="numberPassword"
+        android:maxLength="4" >
+    </EditText>
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:layout_alignParentBottom="true"
+        android:weightSum="2" >
+
+        <Button
+            android:layout_width="wrap_content"
+            android:layout_height="@dimen/min_button_height"
+            android:id="@+id/pin_cancel_button"
+            android:textSize="@dimen/font_size_dp_medium"
```

Comment:
@ctsims So the reason I did this was because a while back I ran into a bug where using sp font sizes was producing really undesirable behavior in the alert dialogs on some devices. This was the PR that fixed that: https://github.com/dimagi/commcare-odk/pull/754. Do you have any idea what might have been the reason for that? I've been hesitant about using sp for font size ever since.